### PR TITLE
Typo fixing multiple sections of the documentation

### PR DESCRIPTION
**Pull request by ma3lstrom**
Closes https://github.com/coala-analyzer/documentation/issues/97

### CODE REVIEW

**Comment by ma3lstrom** [comment]
@sils1297 Please feel free and reject the changes if these commits are removing too many issues for newcomers. I won't mind :)

**Comment by sils1297** [comment]
@Adrianzatreanu can you add a commit section to your coafile? GitMate didn't complain about the commit messages which are clearly too long in some cases

@ma3lstrom thanks so much, I'll review it soon :)

**Comment by Adrianzatreanu** [comment]
Sure I am! Ill also review this in a little bit after i finish some stuff.

File: Developers/Git_Basics.rst [hunk 1]
```diff
@@ -257,7 +257,7 @@ Squashing your commits
 -------------------------
 
 It's possible that you have more than one commit and you want them to be
-squashed into single commit. You can take your series of commits and squash
+squashed into a single commit. You can take your series of commits and squash
```
Comment (Asnelchristian) [inline Developers/Git_Basics.rst]: can you do this in a different commit? :sweat_smile:

File: Developers/Git_Basics.rst [hunk 2]
```diff
@@ -274,7 +274,7 @@ An editor will be fired up with all the commits in your current branch
 (ignoring merge commits), which come after the given commit. Keep the first one
 as "pick" and on the second and subsequent commits with "squash". After saving,
 another editor will be fired up with all the messages of commits which you want
-to squash. Clean up all the messages and add a new message which you want to be
+to squash. Clean up all the messages and add a new message to be
```
Comment (Asnelchristian) [inline Developers/Git_Basics.rst]: this should go in another commit too i think

Comment (Asnelchristian) [inline Developers/Git_Basics.rst]: about your commit message uhm.. what about `Git_Basics: Change 'to'->'the'`? :sweat_smile:, so that you have another commit that adds `a` and another one that remove `which you want`

File: Developers/Git_Basics.rst [hunk 3]
```diff
@@ -174,7 +174,7 @@ Now you would like to get your commit into the actual master branch. Making
 your changes available to all future users of the project. For this, you will
 have to create a Pull Request. To do this, you will have to go on GitHub, on
 your fork page. You should change the branch to the one you have worked and
-submitted the commit on. Now you can create a Pull Request by clicking
+submitted the commit on. Now you can create a Pull Request by clicking the
```
Comment (Asnelchristian) [inline Developers/Git_Basics.rst]: use `Git_Basics` not `Git Basics.rst` in your commit' shortlog 
and maybe `Add 'the'` simply ... :sweat_smile:

**Comment by Asnelchristian** [comment]
maybe you should split https://github.com/coala-analyzer/documentation/pull/108/commits/08db4b7e01f9cf2d9a7012d788393908b380650d into different commits ?!

**Comment by Asnelchristian** [comment]
there is no need of using `.rst` in you commits shortlog

**Comment by Asnelchristian** [comment]
btw. this is amazing

**Comment by ma3lstrom** [comment]
Thanks for the review @Asnelchristian, I'll break the Git_Basics.rst file into separate smaller commits and fix up the commit messages

**Comment by ma3lstrom** [comment]
Same for the External_Bears file

File: Users/Tutorial.rst [hunk 4]
```diff
@@ -28,7 +28,7 @@ use with this sample code and may need minor adjustments.
 Let's Start!
 ------------
 
-There are two options how to let coala know what kind of analysis it
+There are two options used to let coala know what kind of analysis it
```
Comment (sils1297) [inline Users/Tutorial.rst]: this is weird grammarly...

**Comment by sils1297** [comment]
ack https://github.com/coala/documentation/pull/108/commits/218717e375c9c9e85ab4e1530c6e53407cb82eb2 https://github.com/coala/documentation/pull/108/commits/bc3c46d505e024f848031635b5fee06042f62018 https://github.com/coala/documentation/pull/108/commits/19b816d801c2ddfd0ab891c41eb053b340d0555d https://github.com/coala/documentation/pull/108/commits/0095a43832fecf94e46aa6fb5f85046febf37254 https://github.com/coala/documentation/pull/108/commits/31ea8612603601f99265d89b3afd687d604b9d38 https://github.com/coala/documentation/pull/108/commits/7836e604c4a41360cb9dcf86c67ef6b0b40b547b

**Comment by sils1297** [comment]
ack https://github.com/coala/documentation/pull/108/commits/3b25da908b1dd2e088a29fb871ca499f0e9766b1 https://github.com/coala/documentation/pull/108/commits/6c964c790cf6312f461ae0d90c2ee8aba14356d2 https://github.com/coala/documentation/pull/108/commits/4e20969eff27fd00471b06017069bfd21f887921 https://github.com/coala/documentation/pull/108/commits/f3990962dead2cda17264c63f53d2f65dbaf93ec

**Comment by sils1297** [comment]
unack https://github.com/coala/documentation/pull/108/commits/ed78f4c2d0400b9007c0ee9dcd3c86f9f5740252

File: Users/Tutorial.rst (hunk 4 above)
Comment (ma3lstrom) [inline Users/Tutorial.rst]: You're right, the full sentence sounds awkward. How about: 
"coala has two options for code analysis" ?

Comment (sils1297) [inline Users/Tutorial.rst]: I actually don't understand this sentence at all and what it is about :/

Comment (sils1297) [inline Users/Tutorial.rst]: ah I think what he means is that you can use the CLI and config files... I feel like we should completely rework this tutorial it seems to be in a bad state :/

**Comment by sils1297** [comment]
ack https://github.com/coala/documentation/pull/108/commits/d9991f956a73d17ca7ae41d250f379092617bc5b https://github.com/coala/documentation/pull/108/commits/6e73e21998d2bbf36cfdcff7e8cfe77979bac3cb https://github.com/coala/documentation/pull/108/commits/468c36828de67fc383db9f613704c886b080f2d3 https://github.com/coala/documentation/pull/108/commits/606c1048ccca7364584267c586b5dbad1a22a3ee https://github.com/coala/documentation/pull/108/commits/f8a63741390d009431063c31e2502bb75b7874b7 https://github.com/coala/documentation/pull/108/commits/7e5aa5ed100b8270925c69c56ccf2efc5254d87b

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
Hi,

it looks like your PR is out of date and needs a rebase. [This page](http://coala.io/git) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original documentation repository as a remote:

```
git remote add upstream https://github.com/coala/documentation.git
```

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

(If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :). Ping us on [our gitter channel](https://coala.io/chat) if you need help!)

**Comment by sils1297** [comment]
If you don't change your commits while rebasing, please do a `git log --oneline upstream/master..` paste it into a comment here on the PR with a `reack` *right in the beginning of the comment*. If you change commits during rebase remove those SHAs from the comment so they will be re-reviewed

**Comment by ma3lstrom** [comment]
reack 
* 4601654 (upstream/master) .coafile: Add ``invalidlinks`` section
* aba7edc .coafile: Add ``commit`` section
* 13d09e3 .coafile: Change ``README.md``->``README.rst``
* 4b0fbe5 FAQ.rst: Fix broken links
* 3ca120d Provide more instructions for using Docker image
* 5c8c616 Codestyle.rst : Fix typo in codestyle file
* d491d70 Bear_Installation_Tool: Update to reflect the reworked tool
* 416eeda Writing_Bears: Rename 'get_dependencies' to 'BEAR_DEPS'
* 5852ad9 Getting_In_Touch: Add information about issue discussions
* 4af556a Codestyle.rst: Add line continuation policy
* 0363ccd .gitignore: Add *.orig
* 891a78a Add Bear_installation_Tool to index.rst
* 9785208 Add docs for Bear Installation Tool
* 43f8111 Writing_Good_Commits: Remove the word "the"
* 8d7e509 .gitignore: Ignore venv/
* 0a5b7d0 Makefile: Make html section default
* 62c658c Writing_Documentation.rst: Change instruction for new pages
* 9b3dfc4 Improve README.rst by adding build instructions
* 7107d2a Add kinds of bears to docs
* 30291fd Add coafile and fix formatting mistakes
* 0929ff5 Remove italic from ``coala``
* 52f3fd6 Rework Writing_Tests
* 2cc96da Writing_Bears.rst: Fix relative link
* 771cf3c Rework Writing_Documentation
* 777c72e Testing_Bears.rst: Remove italic from ``coala``
* a4d538f Rework Self_Correcting_Bears
* ef7a858 Writing_Linter_Bears.rst: Remove italic from ``coala``
* 03a17ae Rework Writing_Bears
* 8ef8754 Development_Setup.rst: Remove italic from ``coala``
* bbe7979 Rework Review.rst
* c3f65e5 Rework Git_Basics.rst
* 67c25af Rework Codestyle.rst
* a13a9e3 Rework Writing_Good_Commits.rst
* 9b24e54 Tutorial.rst: Fix title underline
* dc6105e Newcomers_Guide: Add note regarding CI builds
* f7ea65f Tutorial.rst: Fix ``seperated``->``separated``
* eb753ce Rework Newcomers_Guide.rst
* 7a27d1b Rework Docker_Image.rst
* 8c29c18 Rework Shell_Autocompletion.rst
* 136157a Rework coala_as_Git_Hook.rst
* f917c95 Rework External_APIs.rst
* 500aa45 Exit_Codes: Remove italic from ``coala``
* 0098217 Glob_Patterns: Remove italic from ``coala``
* b661931 Add External_Bears.rst
* 16bd3c1 MAC_Hints: Add uppercase to title word
* 57c06ca Rework coafile.rst
* 9f2bc26 docs: Update titles correctly to match toctree
* 474451d Rework Tutorial.rst
* 20238d1 Rework Install.rst
* 726d490 Add docs-requirements.txt
* 5d06215 Rework Getting_In_Touch
* 428b594 FAQ: Fix ``docummented``->``documented``
* a7ff17f Getting_In_Touch: Move to Help
* bad9450 index.rst: Rework Welcome page
* d4763fa Restructure file hierarchy
* a0f45bd Fix version to 0.8.0
* dea6eba .gitignore: Ignore _build/
* e2a1139 Move docs from coala
* 8400586 .rultor.yml: Add empty script
* c18b709 Add rultor
* b95263b Initial commit

**Comment by ma3lstrom** [comment]
reack
6437794 External_Bears: Switch 'there' to 'it'
23c77a8 External_Bears: Remove 'it'
f0b01e8 External_Bears: Switch 'go with'
da5f94b Git_Basics: Add 'the'
209279a Git_Basics: Remove 'which...'
9578914 Git_Basics: Add 'a'
c964579 External_Bears: Add missing 'a'
9ae1b7f Self_Correcting_Bears: Add missing 'documentation' word
1e647ab Writing_Linter_Bears: Switch order for 'add' and 'also'
88e64c6 Review: Change word 'There' to 'These'
4d4402f Git_Basics: Change 'to' -> 'the

**Comment by sils1297** [comment]
@rultor merge
